Vegetarian-Food-Finder/Controller/DetailsViewController.swift

```diff
@@ -19,25 +18,73 @@ class DetailsViewController: UIViewController {
 
     @IBOutlet var category: UILabel!
 
+    @IBOutlet var phoneNumberLabel: UIButton!
+
     override func viewDidLoad() {
         super.viewDidLoad()
         configureRestaurnat()
+        let doubleTap = UITapGestureRecognizer(target: self, action: #selector(didDoubleTap(_:)))
+        doubleTap.numberOfTapsRequired = 2
+        imageOfResturant.addGestureRecognizer(doubleTap)
     }
 
+    @IBOutlet var likeBtn: UIButton!
+
     @IBAction func didTapLike(_ sender: Any) {
     }
 
+    func formattedAddress() -> String {
+        let formatter = CNPostalAddressFormatter()
+        let address = CNMutablePostalAddress()
+        guard let restaurant = selectedRestaurant else {
+            return ""
+        }
+        address.street = (restaurant.location.address1)
+        address.postalCode = (restaurant.location.zipCode)
+        address.city = (restaurant.location.city)
+        address.country = (restaurant.location.country)
+        return formatter.string(from: address)
+    }
+
     private func configureRestaurnat() {
-        nameOfResturant.text = selectedRestaurant?.name
-        ratingOfResturant.text = "\(selectedRestaurant?.rating)"
-        let address = selectedRestaurant?.location.address1
-        let state = selectedRestaurant?.location.state
-        let city = selectedRestaurant?.location.city
-        let zipCode = selectedRestaurant?.location.zipCode
-        let country = selectedRestaurant?.location.country
-        resturantAddress.text = "\(address),\(city),\(zipCode),\(state),\(country)"
-        imageOfResturant.load(url: selectedRestaurant!.imageUrl)
-        reviewCountOfRestaurant.text = "\(selectedRestaurant?.reviewCount)"
-        category.text = selectedRestaurant?.categories?[1].title
+        guard let restaurant = selectedRestaurant else {
+            return
+        }
+        nameOfResturant.text = restaurant.name
+        ratingOfResturant.text = "\(restaurant.rating)"
+        resturantAddress.text = formattedAddress()
+        imageOfResturant.load(url: restaurant.imageUrl)
+        reviewCountOfRestaurant.text = "\(restaurant.reviewCount)"
+        category.text = restaurant.categories?[1].title
+        phoneNumberLabel.setTitle("\(restaurant.phone)", for: .normal)
+    }
+
+    @IBAction func phoneNumberPressed(_ sender: Any) {
+    }
+
+    @objc private func didDoubleTap(_ gesture: UITapGestureRecognizer) {
+        guard let gestureView = gesture.view else {
+            return
+        }
+        let size = gestureView.frame.size.width / 4
```

Comment:
nit: Be careful with the naming to avoid confusion. Size usually represents location and measurements. In this case you are getting the length of a side of the frame. Naming it `size` is confusing.